v2/paths.go

```diff
@@ -22,33 +22,26 @@ import (
 	"strings"
 )
 
-// GroupPath is a string that appears as the third field in /proc/PID/cgroup.
-// e.g. "/user.slice/user-1001.slice/session-1.scope"
-//
-// GroupPath must not contain "/sys/fs/cgroup" prefix.
-// GroupPath must be a absolute path starts with "/".
-type GroupPath string
-
 // NestedGroupPath will nest the cgroups based on the calling processes cgroup
 // placing its child processes inside its own path
-func NestedGroupPath(suffix string) (GroupPath, error) {
+func NestedGroupPath(suffix string) (string, error) {
 	path, err := parseCgroupFile("/proc/self/cgroup")
 	if err != nil {
 		return "", err
 	}
-	return GroupPath(filepath.Join(string(path), suffix)), nil
+	return filepath.Join(string(path), suffix), nil
 }
 
 // PidGroupPath will return the correct cgroup paths for an existing process running inside a cgroup
 // This is commonly used for the Load function to restore an existing container
-func PidGroupPath(pid int) (GroupPath, error) {
+func PidGroupPath(pid int) (string, error) {
 	p := fmt.Sprintf("/proc/%d/cgroup", pid)
 	return parseCgroupFile(p)
 }
 
 // VerifyGroupPath verifies the format of g.
 // VerifyGroupPath doesn't verify whether g actually exists on the system.
```

Comment:
I still think we need godoc to clarify what is the expected format.
Probably we should add an example string like `"/user.slice/user-1001.slice/session-1.scope"`